### ISSUE TITLE
about WebAssembly

### ISSUE THREAD

**Issue by yianoohh0** [dstndstn/astrometry.net#324]
I want to make astrometry.net run on the web through WebAssembly, and I want to compile it based on MakeFile dependencies, so I successfully compiled qfits-an. However, during the compilation of the util using emsdk, there was an error in the number in the bl-nl-sort. c file 
`In file included from ext_src/astrometry.net/util/bl-nl-sort.c:5:
ext_src/astrometry.net/include/astrometry/bl-nl.h:17:9: error: unknown type name 'bl'
ext_src/astrometry.net/util/bl-nl.ctypedef bl nl;:30:1: error: unknown type name 'number'

        ^number* NLF(to_array)(nl* list) {

^`
so can I specify the number as a specific type, such as float

**Comment by dstndstn** [comment]
bl-nl-sort.c does not get compiled directly, it gets included in bl-sort.c 

More broadly: I don't know anything about webassembly, but can you load large files?  The index files are pretty big.  The code loads them with the mmap() system call, is that available?  If not, then you're looking at more significant changes to the code.

**Comment by dstndstn** [comment]
Also: currently, the solve-field command runs multiple python programs to do different processing steps (removelines.py, uniformize.py), and to call the actual astrometry engine (astrometry-engine).  So you either need something like exec(), or to link the astrometry-engine directly.  (The use of those python programs can be turned off)

**Comment by yianoohh0** [comment]
My command to execute is as follows:
solve-field  --fits-image %s --overwrite --scale-units degwidth --radius 10 --nsigma 12 --scale-low 7 --scale-high 15 --no-plots --no-remove-lines --uniformize 0 --timestamp -v

I have tracked the execution steps of the code and have not found any calls to Python programs

**Comment by dstndstn** [comment]
Yes, --no-remove-lines and --uniformize 0 turn off those two python programs

**Comment by yianoohh0** [comment]
I once cross compiled Astrometric and ran it on a resource constrained embedded device, which would take a few seconds. So my idea was to run image2xy on the embedded side, calculate the axy file, transfer the axy file to another device's web, and run engine_run_job on the web

**Comment by yianoohh0** [comment]
maybe i can use docker?
